packages/database/src/core/BaseDatabase.js

```diff
@@ -456,6 +456,10 @@ export class BaseDatabase {
   commitTransaction() {
     return this.connection.commit();
   }
+
+  isWithinTransaction() {
+    return this.connection.isTransaction;
+  }
```

Comment:
This reads more as a property than a method (to me). Do you think this would make more sense?

```suggestion
  get isWithinTransaction() {
    return this.connection.isTransaction;
  }
```

(Admittedly can’t remember the rest of this class, so feel free to just resolve silently if it adds inconsistency!)

Reply:
Agreed, or `checkIsWithinTransaction`